lib/remoteResource.js

```diff
@@ -50,46 +55,83 @@ const createRemoteResources = async (razeeApi, apiKey, subscriptions, clusterId)
           'requests': []
         }
       };
-      resourceTemplate.spec.requests.push(parsed);
+      remoteResourceJson.spec.requests.push(parsed);
+      
+      const secretJson = {
+        'apiVersion': 'v1',
+        'kind': 'Secret',
+        'metadata': {
+          'namespace': NAMESPACE,
+          'name': secretName,
+          'annotations': {
+            'deploy.razee.io/clustersubscription': sub.subscriptionUuid,
+            'deploy.razee.io/clusterid': clusterId
+          },
+          'labels': {
+            'razee/watch-resource': 'lite'
+          }
+        },
+        'data': {
+          'razee-api-org-key': apiKey
+        }
+      };
 
-      const opt = { simple: false, resolveWithFullResponse: true };
 
-      const uri = krm.uri({ name: resourceName, namespace: NAMESPACE });
-      log.debug(resourceName);
-      const get = await krm.get(resourceName, NAMESPACE, opt);
-      if (get.statusCode === 200) {
-        // the remote resource already exists so use mergePatch to apply the resource
-        log.info(`Attempting mergePatch for an existing resource ${uri}`);
-        const mergeResult = await krm.mergePatch(resourceName, NAMESPACE, resourceTemplate, opt);
-        if (mergeResult.statusCode === 200) {
-          log.info('mergePatch successful', { 'statusCode': mergeResult.statusCode, 'statusMessage': mergeResult.statusMessage });
-        } else {
-          log.error('mergePatch error', { 'statusCode': mergeResult.statusCode, 'statusMessage': mergeResult.statusMessage });
-        }
-      } else if (get.statusCode === 404) {
-        // the remote resource does not exist so use post to apply the resource
-        log.info(`Attempting post for a new resource ${uri}`);
-        const postResult = await krm.post(resourceTemplate, opt);
-        if (postResult.statusCode === 200 || postResult.statusCode === 201) {
-          log.info('post successful', { 'statusCode': postResult.statusCode, 'statusMessage': postResult.statusMessage });
-        } else {
-          log.error('post error', { 'statusCode': postResult.statusCode, 'statusMessage': postResult.statusMessage });
-        }
-      } else {
-        log.error(`Get ${get.statusCode} ${uri}`);
-      }
+      await applyResource(remoteResourceJson);
+      await applyResource(secretJson);
     }));
 
   } catch (error) {
     log.error('There was an error creating remote resources', { error });
   }
+};
 
+const applyResource = async (resourceJson) => {
+  const resourceName = objectPath.get(resourceJson, 'metadata.name', '');
+  const resourceNamespace = objectPath.get(resourceJson, 'metadata.namespace', NAMESPACE);
+  const resourceApiVersion = objectPath.get(resourceJson, 'metadata.apiVersion', RR_API_VERSION);
+  const resourceKind = objectPath.get(resourceJson, 'metadata.kind', '');
```

Comment:
apiVersion and kind are top level items: see example above for the secret https://github.com/razee-io/ClusterSubscription/pull/130/files#diff-ad5535a8a0ebe356893dafbfa1d42ed0b0df6c72c6d520c840318b55adc3025fR61-R62
```suggestion
  const resourceApiVersion = objectPath.get(resourceJson, 'apiVersion', RR_API_VERSION);
  const resourceKind = objectPath.get(resourceJson, 'kind', '');
```